### ISSUE TITLE
carouselDidEndScrollingAnimation fired at creation

### ISSUE THREAD

**Issue by MarkMolina** [nicklockwood/iCarousel#593]
carouselDidEndScrollingAnimation is getting called when I create the carousel. Since I'm detecting index changes there I don't want this to be fired at creation.

didSelectItemAtIndex will not suffice since that doesn't get fires when a user drags to a different index.

Also I think this library would really improve if didSelectItemAtIndex also got called when a user changes the index by dragging instead only of tapping.

**Comment by nicklockwood** [comment]
Did you try carouselCurrentItemIndexDidChange?

**Comment by MarkMolina** [comment]
Thanks for the quick reply.
That one however gets fired during dragging as well.

**Comment by nicklockwood** [comment]
I thought you said you wanted an event that would get fired both when dragging and when tapping? If not, what is it that you do want?

**Comment by MarkMolina** [comment]
Oke let me rephrase.

Basically I want it to work just like 
- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component;

So there are 2 ways to select an item
- tapping
- dragging to a view

Tapping can be detected with didSelectItemAtIndex. Dragging a view to it's end position doesnt trigger this.

carouselCurrentItemIndexDidChange however fires during dragging as well but I'm only interested in the end position of course.

**Comment by nicklockwood** [comment]
OK, I see. In that case, the best option is probably to make a note of the starting index when you first load the carousel, and then use carouselDidEndScrollingAnimation, and ignore it if the index hasn't changed.

If the user wants to select the item at index zero, they would have to drag away from zero and then drag back to fire the event - but that problem also exists with UIPickerView.

**Comment by MarkMolina** [comment]
Did something similar by ignoring the first request for carouselDidEndScrollingAnimation (the one begin called after creation).

However I still think it's worth taking a look at how didSelectRow for UIPickerView works and implementing didSelectItemAtIndex in the same way. It's up to you in the end though.

Thanks for the update.

**Comment by nicklockwood** [comment]
The thing is, didSelectItemAtIndex doesn't mean the same thing in iCarousel as it does in UIPickerView. In iCarousel, you can select an item that isn't at the current index. It's more like didSelectRow in UITableView.

**Comment by briananderson1222** [comment]
What's the advantage of being able to distinguish when an item is 'selected' from a drag or a click? I would expect the didSelectItemAtIndex to fire when doing a drag event as well which is sadly not the case

**Comment by MarkMolina** [comment]
In the end we're just interested in what the user selected. Dragging, swiping and deaccelerating, or tapping an item, it doesn't really matter. One delegate should handle this.

Makes sense it's didSelectItemAtIndex.

**Comment by nicklockwood** [comment]
There's already a delegate method to tell you when the currently centered index has changed (carouselCurrentItemIndexDidChange:) which will fire regardless of whether it changed because of a swipe or a tap.

Tapping an item is not the same thing as swiping the carousel. I should probably have called didSelectItemAtIndex: didTapItemAtIndex: instead, but I was trying to be consistent with UITableView. That's what it means anyway.

**Comment by MarkMolina** [comment]
Yes but that method gets called on each change. So dragging from index 0 to 5 will trigger multiple times which is of course not what we want.

**Comment by hb2708** [comment]
I agree to https://github.com/nicklockwood/iCarousel/issues/593#issuecomment-90611238 

As it would do your work.
